libethash-cl/CLMiner.cpp

```diff
@@ -858,6 +854,26 @@ bool CLMiner::initEpoch_internal()
                 m_dag.push_back(
                     cl::Buffer(m_context[0], CL_MEM_READ_ONLY, (m_epochContext.dagSize) / 2));
             }
+            cllog << "Creating light cache buffer, size: "
+                  << dev::getFormattedMemory((double)m_epochContext.lightSize);
+            m_light.clear();
+            bool light_on_host = false;
+            try
+            {
+                m_light.push_back(
+                    cl::Buffer(m_context[0], CL_MEM_READ_ONLY, m_epochContext.lightSize));
+            }
+            catch (cl::Error const& err)
```

Comment:
You assume **any** thrown error is due to insufficient memory but others may throw like eg. CL_INVALID_BUFFER_SIZE (yes I know is unlikely lightsize is 0) or CL_DEVICE_NOT_AVAILABLE.
Possibly a stronger evaluation of err.err() should be implemented.

Reply:
Good point. I'll filter for CL_OUT_OF_RESOURCES or CL_OUT_OF_HOST_MEMORY/